### PR TITLE
Update to nightly-2018-08-06

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request bumps the version of rust nightly we are using. This requires the following changes:

- New flag name for the output of `objcopy`.
- Some names were changed in the rust packages.
- Usual rustfmt updates.

Overall, pretty minor changes.

The major impetus to updating is `risc-v` support.

### Testing Strategy

todo

### TODO or Help Wanted

This doesn't actually compile. In particular, the launchxl platform does not compile:

```
$ make
   Compiling kernel v0.1.0 (file:///Users/bradjc/git/tock/kernel)
   Compiling cortexm v0.1.0 (file:///Users/bradjc/git/tock/arch/cortex-m)
   Compiling capsules v0.1.0 (file:///Users/bradjc/git/tock/capsules)
   Compiling cortexm4 v0.1.0 (file:///Users/bradjc/git/tock/arch/cortex-m4)
   Compiling cc26x2 v0.1.0 (file:///Users/bradjc/git/tock/chips/cc26x2)
error[E0080]: this static likely exhibits undefined behavior
   --> /Users/bradjc/git/tock/chips/cc26x2/src/i2c.rs:128:1
    |
128 | pub static mut I2C0: I2CMaster = I2CMaster::new(I2C0REGISTERS);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered undefined bytes at .transfer.val.value.mode.TAG
    |
    = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rust compiler repository if you believe it should not be considered undefined behavior

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.
error: Could not compile `cc26x2`.
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [target/thumbv7em-none-eabi/release/launchxl] Error 101
```

I'm not sure what to do about that.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

**Comment by alevy** [comment]
I think this is relevant: https://github.com/rust-lang/rust/issues/53325

File: libraries/tock-cells/src/map_cell.rs [hunk 1]
```diff
@@ -30,28 +30,24 @@ pub struct MapCell<T> {
...
 // Credit to @japaric: https://github.com/rust-lang/rust/pull/50150
-const unsafe fn uninitialized<T>() -> T {
-    #[allow(unions_with_drop_fields)]
-    union U<T> {
-        none: (),
-        some: T,
-    }
-
-    U { none: () }.some
+#[allow(unions_with_drop_fields)]
```
Comment (ppannuto) [inline libraries/tock-cells/src/map_cell.rs]: The comment above this should be updated, but the rest of this looks fine

**Comment by ppannuto** [comment]
Rebased on latest master and added a commit to bump the tock-registers crate micro version.

----

Do we have a sense for what the implications are of "which nightly" for the library we've published? In this case I think it's fine, as these changes are backwards-compatible with the newer nightly (and if someone is already using a newer nightly, tock-registers won't currently build).

In the future, however, I could imagine a situation where we use a feature introduced by a new nightly. What's the procedure there -- bump version number and document minimum nightly I suppose?

**Pull request merged**